src/views/drivingLicence/check-your-details.html

```diff
@@ -0,0 +1,215 @@
+{% extends "base-form.njk" %}
+{% set hmpoPageKey = "check-your-details" %}
+{% set hmpoH1Class = "govuk-heading-l" %}
+{% from "hmpo-form/macro.njk" import hmpoForm %}
+{% from "hmpo-submit/macro.njk" import hmpoSubmit %}
+{% from "hmpo-radios/macro.njk" import hmpoRadios %}
+{% from "hmpo-warning-text/macro.njk" import hmpoWarningText %}
+{% from "govuk/components/summary-list/macro.njk" import govukSummaryList %}
+
+{% block mainContent %}
+
+    <h1 id="header" data-page="{{hmpoPageKey}}" class="govuk-heading-l">
+        {{ translate("pages.check-your-details.title") }}
+    </h1>
+
+    {{ hmpoWarningText({
+        text: translate("pages.check-your-details.warningText")
+    }) }}
+
+    {% call hmpoForm(ctx, {
+        attributes: {
+            onsubmit: 'window.disableFormSubmit()'
+        }
+    }) %}
+
+        {% if (values.issuedBy == "DVLA") %}
+
+            {{ govukSummaryList({
+                rows: [
+                    {
+                        key: {
+                            text: translate("pages.check-your-details.lastName"),
+                            classes: "govuk-!-font-weight-bold"
+                        },
+                        value: {
+                            text: values.surname
```

Comment:
Can we refactor this slightly so there is less duplication? The only field that will be different in the summary list is the the issueNumber (present in dvla but not dva) So in theory we can just call the if inside the summary list and add the issueNumber Row conditionally on values.licenceIssuer

Reply:
It seems it's not possible to place a conditional within the component itself.
Therefore I've created 3 SummaryLists - the middle SummaryList is wrapped in a conditional so that SummaryList will only show if it's DVLA. 

`{{ govukSummaryList({
            classes: "govuk-!-margin-bottom-0",
            rows: [
                {
                    key: {
                        text: translate("pages.check-your-details.lastName"),
                        classes: "govuk-!-font-weight-bold"
                    },
                    value: {
                        text: values.surname
                    }
                },
                {
                    key: {
                        text: translate("pages.check-your-details.givenNames"),
                        classes: "govuk-!-font-weight-bold"
                    },
                    value: {
                        text: values.firstName + " " + values.middleNames
                    }
                },
                {
                    key: {
                        text: translate("pages.check-your-details.dateOfBirth"),
                        classes: "govuk-!-font-weight-bold $govuk-text-colour"
                    },
                    value: {
                        text: values.dateOfBirth
                    }
                },
                {
                    key: {
                        text: translate("pages.check-your-details.issueDate"),
                        classes: "govuk-!-font-weight-bold"
                    },
                    value: {
                        text: values.issueDate
                    }
                },
                {
                    key: {
                        text: translate("pages.check-your-details.validTo"),
                        classes: "govuk-!-font-weight-bold"
                    },
                    value: {
                        text: values.expiryDate
                    }
                },
                {
                    key: {
                        text: translate("pages.check-your-details.licenceNumber"),
                        classes: "govuk-!-font-weight-bold"
                    },
                    value: {
                        text: values.drivingLicenceNumber
                    }
                }
            ]
        }) }}

        {% if values.licenceIssuer == "DVLA" %}

            {{ govukSummaryList({
                classes: "govuk-!-margin-bottom-0",
                rows: [
                    {
                        key: {
                            text: translate("pages.check-your-details.issueNumber"),
                            classes: "govuk-!-font-weight-bold"
                        },
                        value: {
                            text: values.issueNumber
                        }
                    }
                ]
            }) }}

        {% endif %}

        {{ govukSummaryList({
            rows: [
                {
                    key: {
                        text: translate("pages.check-your-details.postcode"),
                        classes: "govuk-!-font-weight-bold"
                    },
                    value: {
                        text: values.postcode
                    }
                }
            ]
        }) }}`

Visually in the browser, it all looks like the one list because I've set margin-bottom to 0 so there's no gaps between the three SummaryLists.

![Screenshot 2024-10-30 at 10 50 38](https://github.com/user-attachments/assets/e9fa4241-ea6f-4862-a3ab-92f3cf4afd7e)